lib/internal/assert/calltracker.js

```diff
@@ -46,20 +47,23 @@ class CallTracker {
     const callChecks = this.#callChecks;
     callChecks.add(context);
 
-    return function() {
-      context.actual++;
-      if (context.actual === context.exact) {
-        // Once function has reached its call count remove it from
-        // callChecks set to prevent memory leaks.
-        callChecks.delete(context);
-      }
-      // If function has been called more than expected times, add back into
-      // callchecks.
-      if (context.actual === context.exact + 1) {
-        callChecks.add(context);
-      }
-      return ReflectApply(fn, this, arguments);
-    };
+    return new Proxy(fn, {
+      __proto__: null,
+      apply() {
+        context.actual++;
+        if (context.actual === context.exact) {
+          // Once function has reached its call count remove it from
+          // callChecks set to prevent memory leaks.
+          callChecks.delete(context);
+        }
+        // If function has been called more than expected times, add back into
+        // callchecks.
+        if (context.actual === context.exact + 1) {
+          callChecks.add(context);
+        }
+        return ReflectApply(...arguments);
```

Comment:
This syntax relies on mutable `%ArrayIteratorPrototype%.next`, we should avoid using that.
```suggestion
        return ReflectApply(fn, thisArg, argList);
```

Reply:
I don't know about the web standards, but it's on the ES spec: https://tc39.es/ecma262/#sec-runtime-semantics-propertydefinitionevaluation (step 4 of the `PropertyDefinition : PropertyName : AssignmentExpression` section)
AFAIK, there's no proposal to remove it, so I think it's fine as is (and if it was removed from the spec, the tests would start failing, so it won't get unnoticed). I don't mind `ObjectCreate` if you prefer it though.

Reply:
Seem it was moved out of annex b, https://github.com/tc39/ecma262/pull/2125, now NORMATIVE OPTIONAL LEGACY,
Agree with you, I still choose to use `__proto__`.

Reply:
Thanks for the reference, it’s indeed helpful to understand the context around this feature 👍